packages/theme-check-common/src/checks/valid-block-target/index.spec.ts

```diff
@@ -479,6 +500,351 @@ describe('Module: ValidBlockTarget', () => {
+        const theme: MockTheme = {
+          'blocks/image.liquid': '',
+          'blocks/group.liquid': '',
+          'blocks/text.liquid': '',
+          'blocks/slide.liquid': `
+            {% schema %}
+            {
+              "name": "Slide",
+              "blocks": [
+                {
+                  "type": "text"
+                },
+                {
+                  "type": "image"
+                }
+              ]
+            }
+            {% endschema %}
+          `,
+          'sections/slideshow.liquid': `
+            {% schema %}
+            {
+              "name": "Slideshow",
+              "blocks": [
+                {
+                  "type": "slide"
+                }
+              ],
+              "presets": [
+                {
+                  "name": "Default",
+                  "blocks": [
+                    {
+                      "type": "slide",
+                      "blocks": [
+                        {
+                          "type": "group"
+                        }
+                      ]
+                    }
+                  ]
+                }
+              ]
+            }
+            {% endschema %}
+          `,
+        };
+
+        const offenses = await check(theme, [ValidBlockTarget]);
+        expect(offenses).to.have.length(1);
+        const content = theme['sections/slideshow.liquid'];
+        const erroredContent = content.slice(offenses[0].start.index, offenses[0].end.index);
+        expect(erroredContent).to.equal('"group"');
+      });
+
+      it(`should not report errors for nested blocks when they are allowed (${path} bucket)`, async () => {
+        const theme: MockTheme = {
+          'blocks/image.liquid': '',
+          'blocks/text.liquid': '',
+          'blocks/slide.liquid': `
+            {% schema %}
+            {
+              "name": "Slide",
+              "blocks": [
+                {
+                  "type": "text"
+                },
+                {
+                  "type": "image"
+                }
+              ]
+            }
+            {% endschema %}
+          `,
+          'sections/slideshow.liquid': `
+            {% schema %}
+            {
+              "name": "Slideshow",
+              "blocks": [
+                {
+                  "type": "slide"
+                }
+              ],
+              "presets": [
+                {
+                  "name": "Default",
+                  "blocks": [
+                    {
+                      "type": "slide",
+                      "blocks": [
+                        {
+                          "type": "image"
+                        }
+                      ]
+                    }
+                  ]
+                }
+              ]
+            }
+            {% endschema %}
+          `,
+        };
+
+        const offenses = await check(theme, [ValidBlockTarget]);
+        expect(offenses).to.be.empty;
+      });
```

Comment:
Can you add a unit test to verify that this check wont crash when we have a cyclical theme blocks relationship?

Something like this:
blockA has a preset with blockB that nests blockC
blockC has a preset that uses blockA and nests blockB

The thing we're testing is that `await check` does not crash. It should resolve in a reasonable amount of time and also with no offences.

Claude recommended this assert structure 
```typescript
test('resolves within timeout', async () => {
  await expect(async () => {
        const offenses = await check(theme, [ValidBlockTarget]);
        expect(offenses).to.be.empty;
  }).toMatchTimings({
    // Will fail if execution takes longer than 500ms
    maxMs: 500
  })
})
```

Reply:
Sure sure, I can look into this!